### PR TITLE
Project Movies – Axel Hammarbäck

### PR DESCRIPTION
**Pull request by AxelHammarback**
**Link to my deployed project:**
[Let's go to the movies!](http://lets-go-to-the-movies.netlify.app/)

**My reflections on how this week's project turned out:**
Sometimes well, sometimes not so well. Added my thoughts in the readme.md file, check it out if you for some reason are interested in my thoughts.

### CODE REVIEW

**Review by Snabbina — APPROVED** [review]
🏆 Nice work Axel! 

I appreciate your communicative style in your comments and that one gets to take part in your thought process in the README, and to see how you work in figma, you are already a pro! Thanks for charing your knowledge!

And cool that you are trying  out new things like to put in a sticky nav with the next / previous buttons as well.

Clean, clear code and functionality over all, a pleasure to analyze and read it!

Some small optional changes here and there that could be made, but overall approved and upp to at least the basic requirements.

File: code/src/components/BackButton.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import React from 'react'
+
+import { useHistory } from "react-router-dom";
+import arrowLeft from '../assets/ic_24_arrowLeft.svg'
+
+export const BackButton = () => {
+    let history = useHistory();
+    return (
+        <>
```
Comment (EmOlsz) [inline code/src/components/BackButton.js:9]: No need for React Fragment if you return only one element (in your case, `<button />`)

File: code/src/pages/MovieDetails.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+export const MovieDetails = () => {
+  // Why does id need to be in curlies?
+  const { id } = useParams()
+  const API_KEY = "0db2a5ac0c510ce6221cadf605f4d3e6"
+  const MOVIE_URL = `https://api.themoviedb.org/3/movie/${id}?api_key=${API_KEY}&language=en-US`
+  const [movie, setMovie] = useState({})
+  
+
+  console.log(`Id is: ${id}`)
```
[the remark this answers is not in the document]
Comment (EmOlsz) [reply code/src/pages/MovieDetails.js:16]: 👍

File: code/src/pages/MovieDetails.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  const backgroundSize = "original"
+  const posterSize = "w500"
+  const backdropURL = `https://image.tmdb.org/t/p/${backgroundSize}/${movie.backdrop_path}`
+  const posterURL = `https://image.tmdb.org/t/p/${posterSize}/${movie.poster_path}`
+  const imdbURL = `https://www.imdb.com/title/${movie.imdb_id}/`
+  const genres = movie.genres
+
+  return (<>
+    <main
+      className="movie-details-page"
```
Comment (EmOlsz) [inline code/src/pages/MovieDetails.js:33]: No need for line break if we provide only one attribute

File: code/src/pages/MovieList.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+import { MovieCard } from '../components/MovieCard'
+
+export const MovieList = () => {
+  const [pageNumber, setPageNumber] = useState(1)
+  const MOVIES_URL = `https://api.themoviedb.org/3/movie/popular?api_key=0db2a5ac0c510ce6221cadf605f4d3e6&language=en-US&page=${pageNumber}`
+  const [movies, setMovies] = useState([])
+
+  // This one doesn't really work as I intended, but I can't figure out why. You need to press "Next" twice before it actually changes that page.
+  const changePage = (operator) => {
+    console.log(pageNumber)
```
Comment (EmOlsz) [inline code/src/pages/MovieList.js:12]: another console.log. Let's remove all of them

File: code/src/pages/MovieList.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+      fetchAPI()
+    }
+  }
+
+  const fetchAPI = () => {
+    console.log("Inside fetchAPI")
+    fetch(MOVIES_URL)
+      .then(response => response.json())
+      .then(json => setMovies(json.results))
+    // .catch((error) => {
```
[the remark this answers is not in the document]
Comment (EmOlsz) [reply code/src/pages/MovieList.js:41]: Agree

File: code/src/pages/MovieList.js [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+
+import { MovieCard } from '../components/MovieCard'
+
+export const MovieList = () => {
+  const [pageNumber, setPageNumber] = useState(1)
+  const MOVIES_URL = `https://api.themoviedb.org/3/movie/popular?api_key=0db2a5ac0c510ce6221cadf605f4d3e6&language=en-US&page=${pageNumber}`
+  const [movies, setMovies] = useState([])
+
+  // This one doesn't really work as I intended, but I can't figure out why. You need to press "Next" twice before it actually changes that page.
+  const changePage = (operator) => {
```
Comment (EmOlsz) [inline code/src/pages/MovieList.js:11]: Great job with this custom logic, but I think you could achieve same result with `useHistory()` hook from `react-router-dom` package. Check it out!
https://reactrouter.com/web/api/Hooks/usehistory

**Review by AxelHammarback — COMMENTED** [review]
Fixed some stuff to the comments – thanks, Maks and Sabina! 🙌

**Pull request closed without merging**